### PR TITLE
Task/borrar estudio

### PR DESCRIPTION
**Pull request by Ral15**
#### What's this PR do?
This PR includes the US 9.
#### Where should the reviewer start?
at express-app/controllers/estudio.js at line 108 through 134.

#### How should this be manually tested?
Create an estudio and then attempt to delete it.

#### Any background context you want to provide?
The apiToken of the user is hardcoded at the tests, this will change once we can all have the same user for testing.

This template was adapted ~retrived~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/express-app/node_modules/
```
Comment (erikiado) [inline .eslintignore]: haz el PR con develop, son muchos archivos para revisar

File: config.js [hunk 2]
```diff
@@ -2,5 +2,8 @@ module.exports = {
   DEBUG: false,
   //production
   ENV: 'testing',
-  SECRET_SESSION: 'erikiano'
+  SECRET_SESSION: 'erikiano',
+  username: 'eugenio',
+  password: 'erikiado123',
+  apiToken: '082be9526329ebea51d69e11a50217d93f21545d',
 };
```
Comment (erikiado) [inline config.js]: Please specify these are the test_ credentials

File: express-app/routes/urls.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"home": "/",
-	"apiUrl" : "http://127.0.0.1:8000",
+	"apiUrl" : "http://junipero.erikiado.com/",
```
Comment (erikiado) [inline express-app/routes/urls.json]: Does this work or is it beetter without the last /

File: testDB/users.db [hunk 4]
```diff
@@ -1 +1,2 @@
+{"username":"eugenio","password":"erikiado123","apiToken":"082be9526329ebea51d69e11a50217d93f21545d","_id":"V6DEwzQASu52PWI8"}
```
Comment (adael2) [inline testDB/users.db:1]: I think you need to delete the TestDB when you finish your tests. 

And Don't forget to run lint.

File: test/deleteEstudio-test.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      .waitForVisible('#modalContentId')
+      .click('.swal2-confirm')
+      .then(async () => {
+        await sleep(1000);
+        return client.getText('#modalTitleId');
+      })
+      .then(async (modalValue) => {
+        assert.equal(modalValue, '¡Éxito!');
+        await sleep(1000);
+        return client.click('.swal2-confirm');
```
Comment (erikiado) [inline test/deleteEstudio-test.js:181]: Are all these awaits are needed?

File: testDB/users.db (hunk 4 above)
Comment (erikiado) [reply testDB/users.db:1]: this and/or maybe add it to gitignore

File: test/deleteEstudio-test.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+  });
+  /**
+  * Test delete estudio
+  *
+  * This test will delete an estudio
+  */
+  it('should delete an estudio change its status', async function () {
+    const client = this.app.client;
+    return client.setValue('#username',config.username)
+      .setValue('#password', config.password)
```
Comment (erikiado) [inline test/deleteEstudio-test.js:169]: Can't the login be before the tests?

File: test/deleteEstudio-test.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        await sleep(1000);
+        return client.click('.swal2-confirm');
+      })
+      .then(async () => {
+        await sleep(1000);
+        return client.click('#show-deleted');
+      })
+      .then(async () => {
+        await sleep(1000);
+        return client.isVisible('#delete-estudio-' + estudioId);
```
Comment (erikiado) [inline test/deleteEstudio-test.js:189]: Are we not checking it is deleted from DB?

**Review by erikiado — CHANGES_REQUESTED** [review]

File: test/deleteEstudio-test.js (hunk 5 above)
Comment (Ral15) [reply test/deleteEstudio-test.js:181]: yes @erikiado

File: express-app/routes/urls.json (hunk 3 above)
Comment (Ral15) [reply express-app/routes/urls.json]: This works fine, still I'm going to remove the /

File: test/deleteEstudio-test.js (hunk 6 above)
Comment (Ral15) [reply test/deleteEstudio-test.js:169]: I was not able to place the login at the beforeEach()..

File: config.js (hunk 2 above)
Comment (Ral15) [reply config.js]: 👌

File: testDB/users.db (hunk 4 above)
Comment (Ral15) [reply testDB/users.db:1]: 👌

File: test/deleteEstudio-test.js (hunk 7 above)
Comment (Ral15) [reply test/deleteEstudio-test.js:189]: We do not delete estudiso form the DB, we only change its status to 'Eliminado'. That is why I check into the section of 'Eliminados' at the user dashboard for the estudioId.

Do you think it is still necessary to check that in the DB?
@erikiado

**Review by erikiado — APPROVED** [review]

**Pull request merged**